File: Deepseek-Coder-33b-Instruct/stage2/loop_construct_seq_clause.cpp
#include <iostream>
#include <stdlib.h>
#include <openacc.h>

#ifndef T1
//T1:loop construct seq clause,V:2.7-3.3
int test1(){
    int err = 0;
    const int N = 100;
    int a[N], b[N], c[N];

    // Initialize arrays
    for (int i = 0; i < N; ++i) {
        a[i] = i;
        b[i] = i;
        c[i] = 0;
    }

    // Parallelize the loop
    #pragma acc parallel loop
    for (int i = 0; i < N; ++i) {
        c[i] = a[i] + b[i];
    }

    // Sequentially execute the loop
    #pragma acc loop seq
    for (int i = 0; i < N; ++i) {
        if (c[i] != a[i] + b[i]) {
            err = 1;
            break;
        }
    }

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < 1; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: Deepseek-Coder-33b-Instruct/stage2/loop_construct_seq_clause.f90
!T1:loop construct seq clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER, PARAMETER :: N = 100
        INTEGER :: i, sum, sum_expected
        INTEGER, DIMENSION(N) :: array

        ! Initialize array
        DO i = 1, N
          array(i) = i
        END DO

        ! Compute sum using OpenACC seq clause
        sum = 0
        !$acc parallel loop seq
        DO i = 1, N
          sum = sum + array(i)
        END DO
        !$acc end parallel loop

        ! Compute expected sum
        sum_expected = N * (N + 1) / 2

        ! Check if the sums are equal
        IF (sum .ne. sum_expected) THEN
          errors = errors + 1
        END IF

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
        LOGICAL :: test1
        failed = .FALSE.
        failcode = 0
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
        CALL EXIT (failcode)
      END PROGRAM